libs/mrwebrtc-unityplugin/src/native_renderer.cpp

```diff
@@ -133,45 +156,78 @@ void NativeRenderer::DisableRemoteVideo() {
 }
 
 void NativeRenderer::I420ARemoteVideoFrameCallback(
-    void* user_data,
-    const mrsI420AVideoFrame& frame) {
+  void* user_data,
+  const mrsI420AVideoFrame& frame) {
 
-    NativeRenderer* nativeVideo = static_cast<NativeRenderer*>(user_data);
+  // It is possible for one buffer to be empty, each buffer must be checked.
+  if (frame.ydata_ == nullptr || frame.udata_ == nullptr || frame.vdata_ == nullptr) return;
 
-    // RESEARCH: Do we need to keep a frame queue or is it fine to just render
-    // the most recent frame?
+  NativeRenderer* nativeVideo = static_cast<NativeRenderer*>(user_data);
+ 
+  {
+    // Instance lock in case they get updated from UpdateTextures
+    std::lock_guard guard(nativeVideo->m_lock);
+    if ((int)frame.width_ != nativeVideo->m_remoteTextures[0].width || (int)frame.height_ != nativeVideo->m_remoteTextures[0].height) {
+      Log_Warning("NativeRenderer: I420 frame resolution changed from what it was initialized with. Resizing.");
```

Comment:
Do we need a warning here since we have a callback to inform the user? WebRTC changing the video resolution mid-call is a feature and a common action, so it feels weird to log a warning like if something bad or unexpected had happened.
Also, we should check that `g_textureSizeChangeCallback` is not `nullptr`.